src/plugins/controls/public/control_types/time_slider/time_slider_embeddable.tsx

```diff
@@ -0,0 +1,316 @@
+      }
+    });
+  };
+
+  private updateComponentState(changes: Partial<TimeSliderSubjectState>, publish = false) {
+    this.componentState = {
+      ...this.componentState,
+      ...changes,
+    };
+
+    this.internalComponentStateSubject$.next(this.componentState);
+
+    if (publish) {
+      this.componentStateSubject$.next(this.componentState);
+    }
+  }
+
+  private updateInternalOutput(changes: Partial<ControlOutput>, publish = false) {
+    this.internalOutput = {
+      ...this.internalOutput,
+      ...changes,
+    };
+
+    if (publish) {
+      this.updateOutput(this.internalOutput);
+    }
+  }
+
+  private getCurrentDataView$ = () => {
+    const { dataViewId } = this.getInput();
+    if (this.dataView && this.dataView.id === dataViewId)
+      return new Observable<DataView>((subscriber) => {
+        subscriber.next(this.dataView);
+        subscriber.complete();
+      });
+
+    return this.getDataView$(dataViewId);
+  };
+
+  private fetchAvailableTimerange = () => {
+    this.updateComponentState({ loading: true }, true);
+    this.updateInternalOutput({ loading: true }, true);
+
+    const { fieldName, ignoreParentSettings, query, filters, timeRange, ...input } =
+      this.getInput();
+
+    const inputForFetch = {
+      ...input,
+      ...(ignoreParentSettings?.ignoreQuery ? {} : { query }),
+      ...(ignoreParentSettings?.ignoreFilters ? {} : { filters }),
+      ...(ignoreParentSettings?.ignoreTimerange ? {} : { timeRange }),
+    };
+
+    try {
+      this.getCurrentDataView$()
+        .pipe(
+          mergeMap((dataView) => this.fetchRange$(dataView, fieldName, inputForFetch)),
+          take(1)
+        )
+        .subscribe(({ min, max }) => {
+          this.updateInternalOutput({ loading: false });
+          this.updateComponentState({
+            range: {
+              min: min === null ? undefined : min,
+              max: max === null ? undefined : max,
+            },
+            loading: false,
+          });
+        });
+    } catch (e) {
+      this.updateComponentState({ loading: false }, true);
+      this.updateInternalOutput({ loading: false }, true);
+    }
+  };
+
+  public getComponentState$ = () => {
+    return this.componentStateSubject$;
+  };
+
+  public destroy = () => {
+    super.destroy();
+    this.subscriptions.unsubscribe();
+  };
+
+  public reload = () => {
+    this.fetchAvailableTimerange();
+  };
+
+  public render = (node: HTMLElement) => {
+    if (this.node) {
+      ReactDOM.unmountComponentAtNode(this.node);
+    }
+    this.node = node;
+
+    ReactDOM.render(
+      <TimeSliderControlReduxWrapper embeddable={this} reducers={timeSliderReducers}>
+        <TimeSliderComponent
+          componentStateSubject={this.componentStateSubject$}
+          timezone={this.getTimezone()}
+          dateFormat={this.getDateFormat()}
+          fieldName={this.getInput().fieldName}
+        />
+      </TimeSliderControlReduxWrapper>,
```

Comment:
I wonder if there is a way to "snap" to ` 00:00:00.000` of a specific date, or to make (at least until the play button is implemented) the time control slightly less granular? Not sure if it's worth it, since ideally the play button wouldn't make this necessary - but something worth considering 🤷